sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -12,16 +12,108 @@
+      // that universally works in any context in predictable manner. However, it does not mean that
+      // implementation can't use steady_clock in conjunction with this clock: an author can get a
+      // duration beteen two time_points of this clock (or between system_clock::time point ant this
+      // clock's time_point), and add that duration to steady clock's time_point to get a new
+      // time_point in the steady clock's "coordinate system".
+      static constexpr bool is_steady = std::chrono::system_clock::is_steady;
+      static time_point now() noexcept;
+    };
+  } // namespace Details
+
   /**
    * @brief Manages date and time in standardized string formats.
+   * @detail Supports date range from year 0001 to end of year 9999 with 100ns (7 decimal places for
+   * fractional second) precision.
+   * @remark `std::chrono::system_clock::time_point` can't be used, because there is no guarantees
+   * for the date range and precision.
+   * @remark This class is supposed to be able to handle a DateTime that comes over the wire.
    */
-  class DateTime {
+  class DateTime : public Details::Clock::time_point {
+    static DateTime const SystemClockEpoch;
+
   public:
     /**
-     * @brief Units of measurement the difference between instances of @DateTime.
+     * @brief Construct a default instance of @DateTime (00:00:00.0000000 on Janualy 1st, 0001).
      */
-    // 1 == 100 ns (1 / 10,000,000 of a second, 7 fractional digits).
-    typedef std::chrono::duration<int64_t, std::ratio<1, 10000000>> Duration;
+    constexpr DateTime() : time_point() {}
+
+  private:
+    DateTime(
+        int16_t year,
+        int8_t month,
+        int8_t day,
+        int8_t hour,
+        int8_t minute,
+        int8_t second,
+        int32_t fracSec,
+        int8_t dayOfWeek,
+        int8_t localDiffHours,
+        int8_t localDiffMinutes,
+        bool roundFracSecUp = false);
+
+  public:
+    /**
+     * @brief Construct an instance of @DateTime.
+     *
+     * @param year Year.
+     * @param month Month.
+     * @param day Day.
+     * @param hour Hour.
+     * @param minute Minute.
+     * @param second Seconds.
+     *
+     * @throw std::invalid_argument If any parameter is invalid.
+     */
+    explicit DateTime(
+        int16_t year,
+        int8_t month = 1,
+        int8_t day = 1,
+        int8_t hour = 0,
+        int8_t minute = 0,
+        int8_t second = 0)
+        : DateTime(year, month, day, hour, minute, second, 0, -1, 0, 0)
+    {
+    }
+
+    /**
+     * @brief Construct an instance of @DateTime from base class.
+     */
+    constexpr DateTime(time_point const& timePoint) : time_point(timePoint) {}
+
+    /**
+     * @brief Construct an instance of @DateTime from `std::chrono::system_clock::time_point`.
+     * @param systemTime A value of `std::chrono::system_clock::time_point`.
+     */
+    DateTime(std::chrono::system_clock::time_point const& systemTime)
```

Comment:
explicit

Reply:
We discussed it with Jeffrey and decided to make it implicit. Reasons:
1. `DateTime dt = std::chrono::system_clock::now()` - convenient for customers
2. No harm in converting from system_clock - the system_clock's datetime is, realistically, always representable in terms of date range (between year 0001 and 9999; on POSIX system_clock has 1ns precision, while DateTime's is 100ns, but benefits (item 1) seem to outweigh costs.

Reply:
Sure. I brought this up only because we ever got a conclusion in guidelines meeting that constructors should be explicit unless specifically permitted.

Reply:
Thank you, good point, let me to bring this up during our next meeting.
I believe I have justification, since this whole effort with deriving from std::chrono::time_point<Clock> is so that our DateTime mimics chrono as close as possible, so this class stands in the intersection of standard library and Azure SDK integrating them and providing syntactic convenience for our users. Let's see what people say, I hope they agree, but it will be good to get formal approval and confirmation from broader audience, thank you for bringing this up.

Reply:
I also agree it's better to be implicit than explicit for this one, and I see Jeff signed off. So I think it's okay.